pwndbg/disasm/arch.py

```diff
@@ -503,6 +513,37 @@ def _telescope_format_list(self, addresses: List[int], limit: int, emu: Emulator
                 enhance_string_len=enhance_string_len,
             )
 
+    def _enhance_syscall(self, instruction: PwndbgInstruction, emu: Emulator) -> None:
+        if CS_GRP_INT not in instruction.groups:
+            return None
+
+        syscall_register = pwndbg.lib.abi.ABI.syscall().syscall_register
+        syscall_arch = pwndbg.gdblib.arch.name
+
+        # On x86/x64 `syscall` and `int <value>` instructions are in CS_GRP_INT
+        # but only `syscall` and `int 0x80` actually execute syscalls on Linux.
+        # So here, we return no syscall name for other instructions and we also
+        # handle a case when 32-bit syscalls are executed on x64
+        if syscall_register in ("eax", "rax"):
+            mnemonic = instruction.mnemonic
+
+            # We read .imm directly, because at this point we haven't enhanced the operands with values
+            is_32bit = mnemonic == "int" and instruction.operands[0].imm == 0x80
+            if not (mnemonic == "syscall" or is_32bit):
+                return None
```

Comment:
I guess this should go in the `x86` specific disassembly assistant, right? It could do this check and either `return None` or call `super._enhance_syscall`.

Reply:
Good call - making this change.